docs/Reference/Server.md

```diff
@@ -1402,6 +1402,8 @@ handlers. *async-await* is supported as well.
 *Note: If the error `statusCode` is less than 400, Fastify will automatically
 set it at 500 before calling the error handler.*
 
+*Also note* that `setErrorHandler` will ***not*** catch any error inside `onResponse` hook
```

Comment:
We should explain "why": because the response is already sent to the client.

```suggestion
*Also note* that `setErrorHandler` will ***not*** catch any unexpected error inside `onResponse` hook
```

Reply:
I agree to add the reason.
The 'unexpected' could not be true; it will not intercept any error even if you throw an error on purpose, making it 'expected'.

Reply:
Fair

I was thinking that a better place for these information is this docs instead of the `Server.md` one
https://www.fastify.io/docs/latest/Reference/Hooks/#onresponse

Reply:
> I was thinking that a better place for these information is this docs instead of the `Server.md` one https://www.fastify.io/docs/latest/Reference/Hooks/#onresponse

Already added a note here too.